File: frontend/src/types/event.ts
export interface AppEvent {
  id: number;
  title: string;
  type: "SEMINAR" | "WORKSHOP" | "CONFERENCE";
  location: string;
  startTime: string; // ISO string
  endTime: string;
  capacity: number;
  description: string;
  imgUrl: string;
  status: "pending" | "approved" | "rejected";
  reviewedBy?: number;
  ReviewedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEventDTO {
  title: string;
  type: string;
  location: string;
  start_time: string;
  end_time: string;
  capacity: number;
  description: string;
  img_url: string;
}

File: Backend/models/event.go
package models

import "time"

type EventType string

const (
	EventTypeSeminar    EventType = "SEMINAR"
	EventTypeWorkshop   EventType = "WORKSHOP"
	EventTypeConference EventType = "CONFERENCE"
)

type Event struct {
	ID          uint       `json:"id"`
	Title       string     `json:"title"`
	Type        string     `json:"type"`
	Location    string     `json:"location"`
	StartTime   time.Time  `json:"startTime"`
	EndTime     time.Time  `json:"endTime"`
	Capacity    int        `json:"capacity"`
	Description string     `json:"description"`
	ImgUrl      string     `json:"imgUrl"`
	Status      string     `json:"status" gorm:"size:20;default:'pending'"`
	ReviewedBy  *uint      `json:"reviewedBy"`
	ReviewedAt  *time.Time `json:"ReviewedAt"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
}
